docs/documentation/development/bindings/firmware-update.md

```diff
@@ -0,0 +1,118 @@
+
+The `FirmwareUpdateHandler` consists of the following operations:
+* `getThing()`: Returns the thing that is handled by the firmware update handler. 
+* `updateFirmware(org.eclipse.smarthome.core.thing.firmware.Firmware firmware, org.eclipse.smarthome.core.thing.firmware.ProgressCallback progressCallback)`: Updates the firmware for the physical device of the thing that is handled by the firmware update handler. The given progress callback can be used to notify the framework of a new progress step that is going to be executed.
+* `isUpdateExecutable()`: Returns true if the firmware update handler is in a state in which the firmware update for the device can be executed,
+otherwise false (e.g. if the thing is offline or if its status detail is already firmware updating).
+
+The thing returned by the `getThing()` operation must have set the `org.eclipse.smarthome.core.thing.Thing.PROPERTY_FIRMWARE_VERSION` property. If it is not set then the framework cannot determine the thing´s `org.eclipse.smarthome.core.thing.firmware.FirmwareStatus`.
+
+## Implementing the updateFirmware Operation
+
+Initially the implementation should change the thing status to `org.eclipse.smarthome.core.thing.ThingStatus.OFFLINE` and the thing status detail to `org.eclipse.smarthome.core.thing.ThingStatusDetail.FIRMWARE_UPDATING`. Afterwards if the handler would like to notify the framework of update progress information then the sequence of the firmware update progress steps has to be defined using the `defineSequence` operation of the given callback interface. Then by using the `next` operation of the callback interface the handler can indicate that the next progress step is going to be executed.
+
+```java 
+    public void updateFirmware(Firmware firmware, ProgressCallback progressCallback) {
+        // if thing is handled by a thing handler
+        updateStatus(ThingStatus.ONLINE, ThingStatusDetail.FIRMWARE_UPDATING);
+        
+        // define the sequence of the firmware update so that external consumers can listen for the progress
+        progressCallback.defineSequence(ProgressStep.DOWNLOADING, ProgressStep.TRANSFERRING, ProgressStep.UPDATING);
+        
+        // download / read firmware image
+        progressCallback.next();
+        ... read input stream from firmware.getContent();
+        
+        // transfer image to device
+        progressCallback.next();
+        ... transfer image to device
+        
+        // triggering the actual firmware update
+        progressCallback.next();
+        ... trigger update
+        
+        // here: send immediately the success information because it is not mandatory for this implementation to wait for the successful update of the device
+        progressCallback.success();
+        // or: if required to check whether the firmware update was successful or erroneous then wait or spawn a new thread
```

Comment:
I am not sure if we should advice here to spawn a new thread: the FirmwareUpdateHandler can also wait until device has been rebooted, is ONLINE again. get the now current firmware version, which should be the expected one.
All these steps can be done in this method too, or?